flink-dist/src/main/flink-bin/bin/flink

```diff
@@ -17,21 +17,39 @@
 # limitations under the License.
 ################################################################################
 
-target="$0"
 # For the case, the executable has been directly symlinked, figure out
 # the correct bin path by following its symlink up to an upper bound.
 # Note: we can't use the readlink utility here if we want to be POSIX
 # compatible.
-iteration=0
-while [ -L "$target" ]; do
-    if [ "$iteration" -gt 100 ]; then
-        echo "Cannot resolve path: You have a cyclic symlink in $target."
-        break
+followSymLink() {
+    local iteration=0
+    local target=$1
+    while [ -L "$target" ]; do
+        if [ "$iteration" -gt 100 ]; then
+            echo "Cannot resolve path: You have a cyclic symlink in $target."
+            break
+        fi
+        ls=`ls -ld -- "$target"`
+        target=`expr "$ls" : '.* -> \(.*\)$'`
+        iteration=$((iteration + 1))
+    done
+
+    echo "$target"
+}
+
+target=$(followSymLink "$0")
+
+#Check if --configDir is present and set is value as FLINK_CONF_DIR
+if [ "$1" = "--configDir" ]; then
```

Comment:
Here you assume that `--configDir` is the first parameter but later on you parse for it until you find the action command. Why do you set both the environment variable and pass it on as a parameter?